smap/sources/aqiCalculator.py

```diff
@@ -0,0 +1,96 @@
+import requests
+import simplejson as json
+
+# used for logging
+import logging
+import sys, traceback
+logging.basicConfig(level=logging.DEBUG, format='%(asctime)s - %(levelname)s - %(message)s')
+
+BASE_URL = "http://localhost/raqmn_api/"
+
+def init_aqi_drivers():
+	"""
+	Used to initialize AQI Driver
+	Returns all the dylos sensors available with their serial number and uuid
+	"""
+	try:
+		data = json.loads(requests.get(BASE_URL).content)
+		dylosSensors = {}
+		for datum in data:
+			try:
+				pathParts = datum['Path'].split("/") 
+				if len(pathParts)==4 and pathParts[1].strip()=="dylos":
+					dylosSensors[pathParts[2].strip()+"/"+pathParts[3].strip()] = datum['uuid']
+					get_aqi_data(1, datum['uuid'],pathParts[3].strip())
+			except:
+				logging.debug("Unexpected error : "+ str(sys.exc_info()[0]))
+				logging.debug("Traceback : "+ str(traceback.format_exc()))
+		logging.debug(dylosSensors)
+		return dylosSensors
+	except:
+		logging.debug("Unexpected error : "+ str(sys.exc_info()[0]))
+		logging.debug("Traceback : "+ str(traceback.format_exc()))
+        return {}
+
+def get_aqi_data(hours,uuid,type):
+	"""
+	Returns the AQI for number of hours given for the sensor mentioned by uuid
+	"""
+	try:
+		data = json.loads(requests.get(BASE_URL+"data/"+str(hours)+"h/"+uuid+"/").content)
+		dataPoints = remove_outliers([x[1] for x in data])
+		average = float(sum(dataPoints))/len(dataPoints) if len(dataPoints) > 0 else 0
+		aqi = convert_particle_count_to_aqi(average, type)
+		logging.debug(aqi)
+		return aqi
+	except:
+		logging.debug("Unexpected error : "+ str(sys.exc_info()[0]))
+		logging.debug("Traceback : "+ str(traceback.format_exc()))
+        return 0
+
+def remove_outliers(data):
+	"""
+	Removes outliers from timeseries data given
+	TODO: This function uses a very simple trimming algorithm for removing outliers. Need a more robust algorithm for timeseries data to remove outliers
+	"""
+	data.sort()
```

Comment:
Size of data is number of minutes. And we usually calculate average for 1hr (60pts), 24hrs(1440pts) or at max monthly average (43600pts). So in that sense, it looks scalable. Current algorithm though is very trivial, and we can improve, once other things are done.